### DISCUSSION TITLE
FundMe with brownie. maxFeePerGas (0) is less than the block's baseFeePerGas

### DISCUSSION THREAD

**Discussion by dev-Suren** [smartcontractkit/full-blockchain-solidity-course-py#837] [Q&A] [answered]
I have problem while using ganache and brownie. 
During FundMe with brownie lesson if i run the code without running ganache in termal then it wont show any problem but if i run ganache in terminal and run brownie script then it always give me error.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Brownie v1.17.2 - Python development framework for Ethereum

FundmeProject is the active project.
/home/rebootuser/.local/pipx/venvs/eth-brownie/lib/python3.8/site-packages/brownie/network/main.py:44: BrownieEnvironmentWarning: Development network has a block height of 3
  warnings.warn(
Attached to local RPC client listening at '127.0.0.1:8545'...

Running 'scripts/deploy.py::main'...
0x476A84A7c9c000fA9c5fb8Bbf6df42e295690d69
The active network is development
Deploying mocks...
  File "brownie/_cli/run.py", line 50, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/deploy.py", line 25, in main
    deploy_fund_me()
  File "./scripts/deploy.py", line 14, in deploy_fund_me
    deploy_mocks()
  File "./scripts/helpful_scripts.py", line 21, in deploy_mocks
    MockV3Aggregator.deploy(
  File "brownie/network/contract.py", line 528, in __call__
    return tx["from"].deploy(
  File "brownie/network/account.py", line 510, in deploy
    receipt, exc = self._make_transaction(
  File "brownie/network/account.py", line 752, in _make_transaction
    exc = VirtualMachineError(e)
  File "brownie/exceptions.py", line 104, in __init__
    raise ValueError(exc["message"]) from None
ValueError: VM Exception while processing transaction: Transaction's maxFeePerGas (0) is less than the block's baseFeePerGas (586181641) (vm hf=london -> block -> tx)

**Comment by cromewar** [comment]
Hello @dev-Suren if you are already running ganache in terminal that would be a problem with the port, could you please share screenshot showing how you are running it?

**Reply by dev-Suren** [reply]
@cromewar  here is the screen shot.
![Screenshot from 2022-01-22 19-58-09](https://user-images.githubusercontent.com/52071254/150641925-0429ea96-90f4-488c-8299-bd769540aff1.png)

**Comment by cromewar** [comment, marked as answer]
Seems to be a problem with the maxFee your ganache is managing, you can refer [here](https://ethereum.stackexchange.com/questions/62466/how-to-set-maximum-block-gas-in-ganache-cli) to know how to change it.

Anyways when your run test (without having ganache opened before) this will run a instance of ganache and go through the process as you said, also you can try using ganache ui.

btw: which linux distro are you using?

**Reply by dev-Suren** [reply]
Thanks for the help @cromewar. Ganache-ui work perfectly and [https://ethereum.stackexchange.com/questions/62466/how-to-set-maximum-block-gas-in-ganache-cli](url) help me to set the gas limit in galache-cli as well. 

Currently I am running zorin os.

**Reply by HsienChun-Chen** [reply]
Hello @dev-Suren and @cromewar, I have same problem like it. I set gaslimit to 2^53-1, but still can't work.

![image](https://user-images.githubusercontent.com/69665951/150753930-91446188-4f69-4f58-93f5-81e313151020.png)

**Reply by cromewar** [reply]
Your default gas price is still a huge one, I strongly recommend you to reinstall ganache

**Reply by HsienChun-Chen** [reply]
After I reinstall ganache,
It will report an error when ganache executed.
Deploy directly will success.
I wonder know what different between them.
Thanks a lot!

![image](https://user-images.githubusercontent.com/69665951/150785613-1116c87d-79e7-4698-bb2b-77a5c8e08bf4.png)
![image](https://user-images.githubusercontent.com/69665951/150785870-75510307-01cc-4ad7-a2c9-a7e703f91870.png)

**Reply by cromewar** [reply]
I think this might be the way ganache gets executed when you run the command, using ganache cli in terminal makes it persistent as long as you keep it running it, when using ganache with brownie, this creates a new instance of ganache wich start and end at the execution.

**Reply by YassineA2** [reply]
hey, @HsienChun-Chen did you find a solution? i am running into the exact same error but i'm usin WSL, so i dont have access to ganache-ui

**Reply by cromewar** [reply]
Des this: https://stackoverflow.com/questions/70104101/valueerror-method-eth-maxpriorityfeepergas-not-supported-web3-py-with-ganache helps?

**Reply by YassineA2** [reply]
the provided lik is for working with web3.py, not brownie. I don't see how that would fix it
